packages/kbn-config-schema/src/types/type.ts

```diff
@@ -113,7 +113,8 @@ export abstract class Type<V> {
         schema = schema.description(options.meta.description);
       }
       if (options.meta.id) {
-        schema = schema.meta({ [META_FIELD_X_OAS_REF_ID]: options.meta.id });
+        const id = options.meta.id;
+        schema = schema.id(id).meta({ [META_FIELD_X_OAS_REF_ID]: id });
```

Comment:
Leverage Joi's `id` mechanism. Hopefully this does not cause a lot of updates downstream 🤞🏻